### PR TITLE
Coreos buildkite

### PR DESCRIPTION
**Pull request by tank157**
[Pivotal Card](https://www.pivotaltracker.com/story/show/90560272)

Buildkite build has been created, quay.io repo is created and made public.

Assigning @bwreid

### CODE REVIEW

File: script/build [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+else
+  echo "If you'd like to push this to the Docker repo, run: docker push ${DOCKER_REPO}:${IMAGE_TAG}"
+fi
+
+echo '--- updating fleet service template'
+cp $SERVICE_TEMPLATE ${SERVICE}@.service
+
+perl -p -i -e "s|^Environment=DOCKER_REPO=.*$|Environment=DOCKER_REPO=${DOCKER_REPO}|" ${SERVICE}@.service
+perl -p -i -e "s/^Environment=VERSION=.*$/Environment=VERSION=${IMAGE_TAG}/" ${SERVICE}@.service
+perl -p -i -e "s/^Environment=CONTAINER=.*$/Environment=CONTAINER=${SERVICE}/" ${SERVICE}@.service
```
Comment (bwreid) [inline script/build]: Why are we updating the fleet service template file here? I don't doubt it's necessary, just want to understand.

Comment (tank157) [inline script/build]: So the previous way the build and deploy script worked is that the build looked into the (possibly dirty) service file for the docker repo, and then figured out the image tag, and gave you a docker push command with the REPO:IMAGE_TAG. Then if you wanted to deploy locally you pushed that, and then passed that REPO:IMAGE_TAG to the deploy script. The deploy script then changed the service file to reflect those settings, and sent that service file to CoreOS.

This change makes the service file a template, and the build script figures it all out (or you can override with parameters/env vars AT BUILD TIME, and those get backed into the service file which is now a second build artifact. This service file reflects what was built and possible stored up in the docker repo, and the deploy no longer needs to mutate it, it just deploys it. So in a similar way we now back the current configuration into things that get deployed to wildfly (immutable config?) we do the same with the build and the service file.

Comment (bwreid) [inline script/build]: Cool! This helps, thanks.

**Comment by bwreid** [comment]
LGTM

**Pull request merged**